src/test/java/io/vertx/ext/amqp/SenderWithBrokerTest.java

```diff
@@ -202,6 +204,76 @@ public void testSendBasicMessage(TestContext context) {
     });
 
 
+    asyncRecvMsg.awaitSuccess();
+
+  }
+
+  @Test(timeout = 20000)
+  @Repeat(10)
+  public void testDynamicSenderWithOptions(TestContext context) {
+    String address = UUID.randomUUID().toString();
+    String sentContent = "myMessageContent-" + address;
+
+    Async asyncRecvMsg = context.async();
+    AtomicBoolean opened = new AtomicBoolean();
+
+    client = AmqpClient.create(vertx,
+      new AmqpClientOptions().setHost(host).setPort(port).setUsername(username).setPassword(password));
+
+    AtomicReference<AmqpSender> reference = new AtomicReference<>();
+    AtomicReference<Context> exec = new AtomicReference<>();
+    client.connect(res -> {
+      context.assertTrue(res.succeeded());
+
+      res.result().createSender(null,
+        new AmqpSenderOptions()
+        .setDynamic(true)
+        .setLinkName("my-link")
+        .setAutoDrained(false)
+        .setAutoSettle(false),
```

Comment:
None of setLinkName("my-link").setAutoDrained(false).setAutoSettle(false) are really being used or tested here, so I would remove them.